search-setup/indexOperations.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  SearchIndex,
+  SearchIndexClient,
+  SearchField,
+  SearchClient,
+  SearchFieldDataType,
+} from "@azure/search-documents";
+import { Data } from "./types.js";
+
+export async function createSearchIndex(
+  indexClient: SearchIndexClient,
+  indexName: string
+): Promise<void> {
+  const index = buildSearchIndex(indexName);
+  await indexClient.createOrUpdateIndex(index);
+  console.log(`Created or modified index with name: ${indexName}`);
+}
+
+export async function populateIndex(
+  searchClient: SearchClient<Data>,
+  indexData: Data[],
+  indexName: string
+): Promise<void> {
+  await searchClient.mergeOrUploadDocuments(indexData);
+  console.log(`Uploaded data to index with name: ${indexName}`);
+}
```

Comment:
I think you could get the index name from `searchClient` in order to avoid having to pass it in as an argument:
```suggestion
export async function populateIndex(
  searchClient: SearchClient<Data>,
  indexData: Data[]
): Promise<void> {
  await searchClient.mergeOrUploadDocuments(indexData);
  console.log(`Uploaded data to index with name: ${searchClient.indexName}`);
}
```